packages/web3-providers/src/AvatarStore/index.ts

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable unicorn/consistent-function-scoping */
+
+import { memoizePromise } from '@masknet/kit'
+import { memoize } from 'lodash-es'
+import { EnhanceableSite, getEnhanceableSiteType } from '@masknet/shared-base'
+import { type StoreAPI } from '../types/Store.js'
+import { getAddress } from './helpers/getAddress.js'
+import { getAvatar } from './helpers/getAvatar.js'
+import { getAvatarToken } from './helpers/getAvatarToken.js'
+import type { Store, StoreItem } from './types.js'
+import { getPersonaAvatar } from './helpers/getPersonaAvatar.js'
+import { requestIdleCallbackAsync } from '../helpers/requestIdleCallbackAsync.js'
+import { StoreProvider } from '../Store/index.js'
+
+class StoreCreator implements Store {
+    constructor(public items: Map<string, StoreItem | null>) {}
+
+    private retrieve(userId?: string, avatarId?: string, publicKey?: string): StoreItem | null {
+        if (!userId) return null
+        return this.items.get([userId, avatarId, publicKey].join('_')) ?? null
+    }
+
+    retrieveAddress(userId?: string) {
+        return this.retrieve(userId)?.address ?? null
+    }
+
+    retrieveAvatar(userId?: string, avatarId?: string, publicKey?: string) {
+        return this.retrieve(userId, avatarId, publicKey)?.avatar ?? null
+    }
+
+    retrieveToken(userId?: string, avatarId?: string, publicKey?: string) {
+        return this.retrieve(userId, avatarId, publicKey)?.token ?? null
+    }
+}
+
+class AvatarStoreProvider extends StoreProvider<Store> implements StoreAPI.Provider<Store> {
+    public store = new StoreCreator(new Map<string, StoreItem | null>())
+
+    private getAvatarToken = memoizePromise(
+        memoize,
+        (userId: string, avatarId?: string, publicKey?: string): Promise<StoreItem | null> => {
+            return requestIdleCallbackAsync(async () => {
```

Comment:
The result of the callback function doesn't need to be used immediately.

Reply:
> The result of the callback function doesn't need to be used immediately.

sorry can you clarify? I don't understand

Reply:
I created a data hub to store avatar records. It accepts data fetch requests from UI in various places. When it receives a request, it fetches data when the browser is idle, because the request might be quite huge. I don't wanna this task block any UI.

Reply:
but you are not queueing them. so it will block the UI next time you have idle